explainer.md

```diff
@@ -211,13 +216,20 @@ mediaSource.onsourceopen = async (e) => {
 ```javascript
 const mime = 'video/webm; codecs="opus, vp09.00.10.08"';
 const mediaSource = new MediaSource();
-mediaSource.onsourceopen = (e) => {
+mediaSource.onsourceopen = async (e) => {
   const sourceBuffer = mediaSource.addSourceBuffer(mime);
   
   const transport = new Http3Transport("https://example.com/video");
-  transport.onunidirectionalstream = async (e) => {
-    const chunk = await readStreamUntilFin(e.stream);
-    sourceBuffer.appendBuffer(chunk);
+  const streamsReader = transport.receiveStreams().getReader();
+  while (true) {
```

Comment:
Instead of using a while loop for this would it be cleaner to use a function? I'm not sure, so I'll let you decide.

async function appendDataToBuffer(streamResult) {
  if (streamResult.done) {
    return;
  }
  (async () => {
    const chunk = await readStreamUntilFin(e.stream);
    sourceBuffer.appendBuffer(chunk);
  })
  streamsReader.read().then(appendDataToBuffer(result);
}
streamsReader.read().then(appendDataToBuffer(result));

Reply:
I like the idea of the function.  I moved the while loop to it.